src/main/resources/js/script.js

```diff
@@ -0,0 +1,61 @@
+var map;
+var latlngs;
+var polygon;
+var markers;
+var markerCount;
+
+document.addEventListener("DOMContentLoaded", initialize, false);
+
+ function initialize()
+ {
+	console.log("Initializing map");
+ 	L.mapbox.accessToken = 'pk.eyJ1IjoiYWtzbWtyIiwiYSI6ImNpZmY3bnlncDd6eGtzdWtucjFrY3Z3OGkifQ.2QcYNghIAx0ZsZJVewp8QA';
+	map = L.mapbox.map('map', 'mapbox.streets').setView([40, -74.50], 3);
+
+    markerCount=1;
+	latlngs = new Array();
+	map.on('click', function(event){
+		//addToTable(event.latlng);
+		showPolygon(event.latlng);
+	});
+ }
+
+ function showPolygon(latlng)
+ {
+ 	if(polygon != undefined)
```

Comment:
just right if(polygon) {} because  javascript itself checks for null, undefined and empty you don't need to manually write if(polygon != undefined)